content/rust-compiler-performance-survey-2025-results/index.md

```diff
@@ -0,0 +1,325 @@
+
+One area where clean builds might happen often is Continuous Integration (CI). 1495 respondents said that they use CI to build Rust code, and around 25% of them consider its performance to be a big blocker for them. However, almost 36% of respondents who consider CI build performace to be a big issue said that they do not use any caching in CI, which we found surprising. One explanation might be that the generated artifacts (the `target` directory) is too large for effective caching, and runs into usage limits of CI providers, which is something that we saw mentioned repeatedly in the open answers section. We have recently introduced an experimental Cargo and compiler option called [`-Zembed-metadata`][cargo-no-embed-metadata] that is designed to reduce the size of the `target` directory, which might help with this issue somewhat in the future.
+
+## Debug information
+
+The default Cargo `dev` profile generates full debug information (debuginfo) both for workspace crates and also all dependencies. This enables stepping through code with a debugger, but it also increases disk usage of the `target` directory, and crucially it makes compilation and linking slower. This effect can be quite large, as our benchmarks show a possible improvement of [2-30% in cycle counts][perf-debug-line-tables-only] if we reduce the debuginfo level to `line-tables-only` (which only generates enough debuginfo for backtraces to work), and the improvements are [even larger][perf-debug-false] if we disable debuginfo generation completely[^lld].
+
+[^lld]: This benchmark was already performed using the fast LLD linker. If a slower linker was used, the build time wins would likely be even larger.
+
+However, if Rust developers debug their code after most builds, then this cost might be justified. We thus asked them how often they use a debugger to debug their Rust code:
+
+<!-- Chart debugger start -->
+<div>
+    <div class="bar-chart" id="debugger" style="height:500px; width:100%;"><noscript>
+<img alt="debugger" height="500" src="debugger.png"/>
+</noscript></div>
+    <div style="display: flex; margin-bottom: 10px;">
+        <span>[<a href="debugger.png" target="_href_" title="Download chart as PNG">PNG</a>]</span>&nbsp;<span>[<a href="debugger.svg" target="_href_" title="Download chart as SVG">SVG</a>]</span>
+    </div>
+</div>
+<!-- Chart debugger end -->
+
+Based on these results, it seems that the respondents of our survey do not actually use a debugger all that much[^why-not-debug]. Similarly, only 6% of respondents answered that they use a profiler (which also benefits from at least basic debuginfo) often.
```

Comment:
discussing profilers here feels like a non-sequitor you are unlikely to profile a `dev` built project.

Reply:
When I re-read the section, I agree with you. Debuginfo for optimized builds is not very relevant to what I am talking about here. Removed the mention of profilers (they still are in the PDF report, so people can see the results). Thanks!